HW0/P6/P6.txt

```diff
@@ -1 +1,41 @@
-# Your answers here
+###########################################################################################
+## Problem 6A
+###########################################################################################
+
+When running P6A.py, the 10 jobs did not always finish running in the same order that they started. For example, the code output below shows that the first batch of four jobs started in order 0, 1, 2, 3 but job 1 finished before job 0. Similarly, the second batch of four jobs started in order 4, 5, 6, 7 but job 7 finished before job 6.
+
+This appears to be 'normal' behavior rather than an outlier; in fact, the code had to be run 15 times before getting an output where all 10 jobs ended in the same order that they had started. This implies that it is difficult to guarantee consistent runtimes across individual jobs, even for very simple jobs like this one, and that it is likely to be even harder with larger jobs.
+
+To test this, I increased the sleep time from 0.25 to 1.0.  As expected, this resulted in the jobs being even more out of order with each other: in a few cases, the second or third batches of jobs started running before all the jobs in the previous batch had finished running, and in once instance even the first batch of jobs started out of order. In fact, out of 30 total runs, the jobs did not once end in the same order that they had started. These results indicate that smaller job sizes are necessary when attempting to maintain consistent run times across jobs and/or ordered jobs.
```

Comment:
Good experiment! However, I don't think one should make jobs smaller just to maintain consistent running order. A parallel program is not meant to be running in order! Instead, we just need to ensure that input values to subsequent functions is valid, as you have pointed out in the next paragraph